doc/man/man5/strss7.5.man
'\" rtp
.\" vim: ft=nroff sw=4 noet nocin nosi com=b\:.\\\" fo+=tcqlorn tw=77
.\" =========================================================================
.\"
.\" @(#) $Id$
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2008-2009  Monavacon Limited <http://www.monavacon.com/>
.\" Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
.\" Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to copy, distribute and/or modify this manual
.\" under the terms of the GNU Free Documentation License, Version 1.3 or
.\" any later version published by the Free Software Foundation; with no
.\" Invariant Sections, no Front-Cover Texts, and no Back-Cover Texts.  A
.\" copy of the license is included in the section entitled "GNU Free
.\" Documentation License".
.\"
.\" Permission to use, copy and distribute this manual without modification,
.\" for any purpose and without fee or royalty is hereby granted, provided
.\" that both the above copyright notice and this permission notice appears
.\" in all copies and that the name of OpenSS7 Corporation not be used in
.\" advertising or publicity pertaning to distribution of this documentation
.\" or its contents without specific, written prior permission.  OpenSS7
.\" Corporation makes no representation about the suitability of this manual
.\" for any purpose.  It is provided "as is" without express or implied
.\" warranty.
.\"
.\" Permission is granted to process this file through groff and print the
.\" results, provided the printed document carries a copying permission
.\" notice identical to this one except for the removal of this paragraph
.\" (this paragraph not being relevant to the printed manual).
.\"
.\" OPENSS7 CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS MANUAL
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
.\" PARTICULAR PURPOSE, NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THE
.\" DOCUMENT ARE SUITABLE FOR ANY PURPOSE, OR THAT THE IMPLEMENTATION OF
.\" SUCH CONTENTS WILL NOT INFRINGE ON ANY THIRD PARTY PATENTS, COPYRIGHTS,
.\" TRADEMARKS OR OTHER RIGHTS.  IN NO EVENT SHALL OPENSS7 CORPORATION BE
.\" LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES OR ANY
.\" DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
.\" IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
.\" OUT OF OR IN CONNECTION WITH ANY USE OF THIS DOCUMENT OR THE PERFORMANCE
.\" OR IMPLEMENTATION OF THE CONTENTS THEREOF.
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this
.\" manual page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from
.\" the use of the information contained herein.  The author(s) may not have
.\" taken the same level of care in the production of this manual, which is
.\" licensed free of charge, as they might when working professionally.  The
.\" author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by the
.\" source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software
.\" on behalf of the U.S. Government ("Government"), the following
.\" provisions apply to you.  If the Software is supplied by the Department
.\" of Defense ("DoD"), it is classified as "Commercial Computer Software"
.\" under paragraph 252.227-7014 of the DoD Supplement to the Federal
.\" Acquisition Regulations ("DFARS") (or any successor regulations) and the
.\" Government is acquiring only the license rights granted herein (the
.\" license rights customarily provided to non-Government users).  If the
.\" Software is supplied to any unit or agency of the Government other than
.\" DoD, it is classified as "Restricted Computer Software" and the
.\" Government's rights in the Software are defined in paragraph 52.227-19
.\" of the Federal Acquisition Regulations ("FAR") (or any successor
.\" regulations) or, in the cases of NASA, in paragraph 18.52.227-86 of the
.\" NASA Supplement to the FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.\" Last Modified $Date$ by $Author$
.\"
.\" -------------------------------------------------------------------------
.\"
.\" $Log$
.\" =========================================================================
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database openss7.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
search-truncate 40
search-ignore T
.R2
.so openss7.macros
.\"
.\"
.TH @PACKAGE_UCNAME@ 5 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ System File Formats"
.SH NAME
.B @PACKAGE@
\- @PACKAGE_TITLE@ package
.\"
.\"
.SH OVERVIEW
.PP
This is the
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@-@VERSION@ )
package for
.BR "Linux Fast-STREAMS" .
.[
lfs
.]
.PP
.BR "Linux Fast-STREAMS" ,
.BR streams (5),
consists of a base package that provides basic
.I SVR 4.2 STREAMS
capabilities for
.B Linux
and a number of separate add-on packages.
.PP
OpenSS7
.I STREAMS
add-on packages provide additional functionality and protocol support not traditionally found on
UNIX systems, or which have traditionally fallen into the domain of value-add system software, or,
in this case, where functionality was optional or differently handled in a number of
implementations.
.PP
The
.B @PACKAGE_TITLE@
package is one of the several optional add-on packages, this one that provides support for SS7 and
is part of the initial thrust of the
.IR "OpenSS7 Project" .
.[
%T The OpenSS7 Project
%O http://www.openss7.org
.]
.\"
.\"
.SH DESCRIPTION
.PP
The
.B @PACKAGE_TITLE@
package provides
.I STREAMS
modules and drivers, libraries, programs, init scripts, daemons and agents that implement the
.I STREAMS
SS7 subsystem.
.PP
The package builds and installs the following
.I STREAMS
modules and drivers (as kernel modules):
.IP \(en \w'\(en\(em'u
.B streams_sdl_ch.ko
.PD 0
.IP \(en
.B streams_sdl.ko
.IP \(en
.B streams_sdlm.ko
.IP \(en
.B streams_sdl_pmod.ko
.IP \(en
.B streams_sdt.ko
.IP \(en
.B streams_sdt_pmod.ko
.IP \(en
.B streams_sl.ko
.IP \(en
.B streams_sl_min.ko
.IP \(en
.B streams_sl_mux.ko
.IP \(en
.B streams_slpmod.ko
.IP \(en
.B streams_sl_x100p.ko
.IP \(en
.B streams_sl_x400p.ko
.IP \(en
.B streams_sm_mod.ko
.IP \(en
.B streams_spm.ko
.IP \(en
.B streams_mtp.ko
.IP \(en
.B streams_mtp_min.ko
.IP \(en
.B streams_mtp_mod.ko
.IP \(en
.B streams_mtp_npi.ko
.IP \(en
.B streams_mtp_tpi.ko
.IP \(en
.B streams_sccp.ko
.IP \(en
.B streams_sccp_mod.ko
.IP \(en
.B streams_tcap.ko
.IP \(en
.B streams_tc_mod.ko
.IP \(en
.B streams_tr_mod.ko
.IP \(en
.B streams_isup.ko
.IP \(en
.B streams_x100p-ss7.ko
.IP \(en
.B streams_x400p-ss7.ko
.PD
.PP
The following are
.I STREAMS
modules installed by the
.B @PACKAGE@
package:
.PP
The
.B @PACKAGE_TITLE@
package includes kernel modules,
.I SVR 4.2 STREAMS
drivers, modules, libraries, utilities, test programs, daemons, agents, and development environment
for the development and execution of
.B @PACKAGE_TITLE@
protocol stack components and applications for the
.I SVR 4.2 STREAMS
environment.
.PP
This distribution is only currently applicable to
.B Linux
2.4 and 2.6 kernels and was targeted at
.IR ix86 ", " x86_64 ", " ppc " and " ppc64
architectures, but should build and install for other architectures as well.
.PP
This package represents SS7 components that were previously part of the
.BR strss7 (5)
package, but which were isolated into this package at the end of 2006.  Although this manual page
documents all commercial components in the
.BR strss7 (5)
package, not all components have been released publicly.
.\"
.\"
.SH "PACKAGE CONTENTS"
.PP
The
.BR "@PACKAGE_TITLE@ " ( @PACKAGE@ - @VERSION@ )
package contains the following manual pages:
.HP
.BR @PACKAGE@ (5)\ \-\ 
.B @PACKAGE_TITLE@
package.
.PD 0
.HP
.BR @PACKAGE@_mknod (8)\ \-\ 
create or remove
.I STREAMS
SS7 device nodes.

.HP
.BR xti_mtp (3)\ \-\ 
description of the XTI interface to MTP.
.HP
.BR xti_sccp (3)\ \-\ 
description of the XTI interface to SCCP.
.HP
.BR xti_ss7 (3)\ \-\ 
description of the XTI interface to SS7.
.HP
.BR xti_sscop (3)\ \-\ 
description of the XTI interface to SSCOP.
.HP
.BR xti_tcap (3)\ \-\ 
description of the XTI interface to TCAP.
.\".PD
.\".HP
.\".BR acb56 (4)\ \-\ 
.\".PD 0
.\".HP
.\".BR bssap (4)\ \-\ 
.\".HP
.\".BR bssap_ioctl (4)\ \-\ 
.\".HP
.\".BR ch-sdl (4)\ \-\ 
.\".HP
.\".BR ch_x400p (4)\ \-\ 
.\".HP
.\".BR dl (4)\ \-\ 
.\".HP
.\".BR dlb (4)\ \-\ 
.\".HP
.\".BR dl_gsma (4)\ \-\ 
.\".HP
.\".BR dl_ioctl (4)\ \-\ 
.\".HP
.\".BR fr (4)\ \-\ 
.\".HP
.\".BR gsma (4)\ \-\ 
.\".HP
.\".BR hdlc (4)\ \-\ 
.\".HP
.\".BR hdlc_ioctl (4)\ \-\ 
.\".HP
.\".BR isup (4)\ \-\ 
.\".HP
.\".BR isup_ioctl (4)\ \-\ 
.\".HP
.\".BR lm (4)\ \-\ 
.\".HP
.\".BR lmi_ioctl (4)\ \-\ 
.\".HP
.\".BR mtp (4)\ \-\ 
.\".HP
.\".BR mtp_ioctl (4)\ \-\ 
.\".HP
.\".BR mtp-mgmt (4)\ \-\ 
.\".HP
.\".BR mtp_min (4)\ \-\ 
.\".HP
.\".BR mtp-min (4)\ \-\ 
.\".HP
.\".BR mtp_mod (4)\ \-\ 
.\".HP
.\".BR mtp-mod (4)\ \-\ 
.\".HP
.\".BR mtp-mtpi (4)\ \-\ 
.\".HP
.\".BR mtp_npi (4)\ \-\ 
.\".HP
.\".BR mtp-npi (4)\ \-\ 
.\".HP
.\".BR mtp_tpi (4)\ \-\ 
.\".HP
.\".BR mtp-tpi (4)\ \-\ 
.\".HP
.\".BR mx_x400p (4)\ \-\ 
.\".HP
.\".BR OpenSS7 (4)\ \-\ 
.\".HP
.\".BR pca200e (4)\ \-\ 
.\".HP
.\".BR sccp (4)\ \-\ 
.\".HP
.\".BR sccp-gtt (4)\ \-\ 
.\".HP
.\".BR sccp_ioctl (4)\ \-\ 
.\".HP
.\".BR sccp-mgmt (4)\ \-\ 
.\".HP
.\".BR sccp_mod (4)\ \-\ 
.\".HP
.\".BR sccp-mod (4)\ \-\ 
.\".HP
.\".BR sccp-npi (4)\ \-\ 
.\".HP
.\".BR sccp-sccpi (4)\ \-\ 
.\".HP
.\".BR sccp-tpi (4)\ \-\ 
.\".HP
.\".BR sdl (4)\ \-\ 
.\".HP
.\".BR sdl-acb56 (4)\ \-\ 
.\".HP
.\".BR sdl_ch (4)\ \-\ 
.\".HP
.\".BR sdl_eth (4)\ \-\ 
.\".HP
.\".BR sdl_ioctl (4)\ \-\ 
.\".HP
.\".BR sdlm (4)\ \-\ 
.\".HP
.\".BR sdlm-lm (4)\ \-\ 
.\".HP
.\".BR sdl-mux (4)\ \-\ 
.\".HP
.\".BR sdl_pmod (4)\ \-\ 
.\".HP
.\".BR sdl-pmod (4)\ \-\ 
.\".HP
.\".BR sdl_rtp (4)\ \-\ 
.\".HP
.\".BR sdl_x400p (4)\ \-\ 
.\".HP
.\".BR sdt (4)\ \-\ 
.\".HP
.\".BR sdt-acb56 (4)\ \-\ 
.\".HP
.\".BR sdt_ioctl (4)\ \-\ 
.\".HP
.\".BR sdt_pmod (4)\ \-\ 
.\".HP
.\".BR sdt-pmod (4)\ \-\ 
.\".HP
.\".BR sdt_x400p (4)\ \-\ 
.\".HP
.\".BR sl (4)\ \-\ 
.\".HP
.\".BR sl-acb56 (4)\ \-\ 
.\".HP
.\".BR sl_ioctl (4)\ \-\ 
.\".HP
.\".BR slm (4)\ \-\ 
.\".HP
.\".BR sl_min (4)\ \-\ 
.\".HP
.\".BR sl-min (4)\ \-\ 
.\".HP
.\".BR sl_mux (4)\ \-\ 
.\".HP
.\".BR slpmod (4)\ \-\ 
.\".HP
.\".BR sl_x100p (4)\ \-\ 
.\".HP
.\".BR sl_x400p (4)\ \-\ 
.\".HP
.\".BR sm_mod (4)\ \-\ 
.\".HP
.\".BR spm (4)\ \-\ 

NEWS
OpenSS7 Master Package -- history of user visible changes.  2009-04-04
$Id: NEWS,v 0.9.2.72 2009-04-19 00:28:25 brian Exp $
Copyright (c) 2008-2009  Monavacon Limited. <http://www.monavacon.com/>
Copyright (c) 2001-2008  OpenSS7 Corporation. <http://www.openss7.com/>
Copyright (c) 1997-2001  Brian Bidulock <bidulock@openss7.org>
See the end for copying conditions (for this file).

The sections that follow provide information on OpenSS7 releases of the
`OpenSS7 Master Package' package.

 Major changes for release openss7-0.9.2.F.20090405
---------------------------------------------------

This is the seventh `OpenSS7 Project' release of the `OpenSS7 Master
Package' package.  For a sub-package listing, see `README'
(http://www.openss7.org/codefiles/openss7/README) in the release.  Most
of the sub-packages in the release are production grade for `Linux
Fast-STREAMS'.(1)  All existing validation test suites run clean on
supported distributions, production kernels and architectures.

The original intention was that the `OpenSS7 Master Package' package
not be released as often as subpackages and, indeed, newer subpackages
can be strapped into the master package and built automatically.
However, in reality, the `OpenSS7 Master Package' package has been
released as often as the subpackages.  To rebuild the master package
with a new sub-package release, simply replace the directory to which
the sub-package belongs with the unpackged sub-package reelase and then
rebuild the master package.  This release provides support for recent
distributions, production kernels and tool chains.  The release also
provides support for significiant new features in the sub-packages.

Major features since the last public release are:

   - Addition of the `stratm' sub-package to the master build and
     release that now includes all of the ATM components that were
     previously part of the `strss7' sub-package.

   - Significant rework of the `strchan' sub-package into the `strss7',
     `strisdn', `strx25' and `striso' packages.  One single driver now
     meets the needs of all sub-packages.

   - Completion of the `strsock' package, moves from pre-alpha to beta.

   - The `strnsl' subpackage moves from alpha to production.

   - Significant completion of the `sigtran' package: all components
     moved back to production.


_This is a public, stable, production grade release of the package: it
deprecates previous releases.  Please upgrade to the current release
before reporting bugs._

As with other OpenSS7 releases, this release configures, compiles,
installs and builds RPMs and DEBs for a wide range of Linux 2.4 and 2.6
`rpm(8)' and `dpkg(1)' based distributions, and can be used on
production kernels without patching or recompiling the kernel.

This package is publicly released strictly under the `GNU Affero
General Public License Version 3'; however, commercial licensing is
available.  The release is available as an `autoconf(1)' tarball, SRPM,
DSC, and a set of binary RPMs and DEBs for popular distributions.  See
the package page (http://www.openss.org/openss7_pkg.html) for the
`autoconf(1)' tarballs, SRPMs, DSCs and binary RPMs and DEBs.  See the
repository page (http://www.openss7.org/repos.html) for information
concerning network installation and update sources.

See
`http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/ChangeLog'
and `http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/NEWS' in
the release for more information.  Also, see the `openss7.pdf' manual
in the release (also in html
(http://www.openss7.org/openss7_manual.html)).

For the news release, see `http://www.openss7.org/rel20081029_L.html'.

---------- Footnotes ----------

(1) Sub-packages are no longer supported on the deprecated `LiS' STREAMS
package.

 Major changes for release openss7-0.9.2.G
------------------------------------------

This is the sixth public release of the OpenSS7 Master Package.  See
`README' in the release for a sub-package listing.  Most of the
sub-packages in the release are production grade for `Linux
Fast-STREAMS'.  All existing validation test suites run clean on
supported distributions and architectures.

The `OpenSS7 Master Package' is not released as often as the
sub-packages.  As sub-packages are released more often, to rebuild the
master package with a new sub-package release, simply replace the
directory to which the sub-package belongs with the unpacked
sub-package release and then rebuild the master package.  This release
provides support for recent distributions and tool chains.

Major features since the last public release are as follows:

   - License upgrade to `AGPL Version 3'.

   - Modifications to build under `Fedora' `2.6.22.5-49' kernel.  These
     changes also support `2.6.22.9-91.fc7' kernel.  Modifications to
     build under `Fedora' `2.6.25-45.fc9' and `2.6.26.5-45.fc9' kernels.
     Documented `lib32gcc1' problem on Ubuntu.  Noted problem running
     under XEN kernels.  XEN kernels are not yet supported.  Added
     `MODULE_VERSION' to all modules and drivers.

   - Ability to strap out major documentation build and installation
     primarily for embedded targets.  Improvements to common build
     process for embedded and cross-compile targets.  Cross-compile
     fixes (strap out `AC_FUNC_REALLOC' macro when cross-compiling).
     Conversion of RPM spec files to common approach for major
     subpackages.

     Build system now builds `yum(8)' repositories for RPMs and

src/util/init/strss7.sh
#!/bin/sh
#
# @(#) $RCSfile$ $Name$($Revision$) $Date$
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
# All Rights Reserved.
#
# Distributed by OpenSS7 Corporation.  See the bottom of this script for copying
# permissions.
#
# These are arguments to update-rc.d ala chkconfig and lsb.  They are recognized
# by openss7 install_initd and remove_initd scripts.  Each line specifies
# arguments to add and remove links after the the name argument:
#
# strss7:	start and stop strss7 subsystem
# update-rc.d:	start 33 S . stop 33 0 6 .
# config:	/etc/default/strss7
# probe:	false
# hide:		false
# license:	GPL
# description:	This STREAMS init script is part of Linux Fast-STREAMS.  \
#		It is responsible for ensuring that the necessary STREAMS \
#		character devices are present in the /dev directory and \
#		that the STREAMS SS7 subsystem is configured and loaded.
#
# LSB init script conventions
#
### BEGIN INIT INFO
# Provides: strss7
# Required-Start: streams $network
# Required-Stop: streams $network
# Default-Start: 3 4 5
# Default-Stop: 0 1 2 6
# X-UnitedLinux-Default-Enabled: yes
# Short-Description: start and stop STREAMS SS7 subsystem
# License: GPL
# Description:	This STREAMS SS7 init script is part of Linux Fast-STREAMS.
#	It is reponsible for ensuring that the necessary STREAMS SS7 character
#	devices are present in the /dev directory and that the STREAMS SS7
#	subsystem is configured and loaded.
### END INIT INFO

PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
name='strss7'
config="/etc/default/$name"
desc="the STREAMS SS7 subsystem"
mknod="${name}_mknod"

[ -e /proc/modules ] || exit 0

if test -z "$STRSS7_MKNOD" ; then
    for STRSS7_MKNOD in /sbin/${mknod} /usr/sbin/${mknod} /bin/${mknod} /usr/bin/${mknod} ; do
	if [ -x $STRSS7_MKNOD ] ; then
	    break
	else
	    STRSS7_MKNOD=
	fi
    done
fi

# Specify defaults

[ -n "$STRSS7_PRELOAD"       ] || STRSS7_PRELOAD=""
[ -n "$STRSS7_DRIVERS"       ] || STRSS7_DRIVERS="streams-sl-x400p"
[ -n "$STRSS7_MODULES"       ] || STRSS7_MODULES=""
[ -n "$STRSS7_MAKEDEVICES"   ] || STRSS7_MAKEDEVICES="yes"
[ -n "$STRSS7_REMOVEDEVICES" ] || STRSS7_REMOVEDEVICES="yes"

# Source config file
for file in $config ; do
    [ -f $file ] && . $file
done

[ -z "$STRSS7_MKNOD" ] && STRSS7_MAKEDEVICES="no"
[ -z "$STRSS7_MKNOD" ] && STRSS7_REMOVEDEVICES="no"

RETVAL=0

umask 077

case :$VERBOSE in
    :no|:NO|:false|:FALSE|:0|:)
	redir='>/dev/null 2>&1'
	;;
    *)
	redir=
	;;
esac

build_options() {
    # Build up the options string
    :
}

start() {
    echo -n "Loading STREAMS kernel modules: "
    RETVAL=0
    modules=
    for module in $STRSS7_PRELOAD ; do
	modules="${modules:+$modules }$module"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if ! eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -k -q -- $module $redir"
	    [ $? -eq 0 ] || echo -n "(failed)"
	fi
    done
    echo "."

    echo -n "Starting $desc: $name "
    build_options
    if [ $? -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
	RETVAL=1
    fi

    if eval "grep '^[[:space:]]*${name}[/.]' /etc/sysctl.conf $redir" ; then
	echo -n "Reconfiguring kernel parameters: "
	eval "sysctl -p /etc/sysctl.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	fi
    fi

    if [ -f /etc/${name}.conf ] ; then
	echo -n "Configuring STREAMS parameters: "
	eval "sysctl -p /etc/${name}.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi

    if [ -n "$STRSS7_MKNOD" -a ":$STRSS7_MAKEDEVICES" = ":yes" ] ; then
	echo -n "Making STREAMS SS7 devices: "
	$STRSS7_MKNOD
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    return $RETVAL
}

stop() {
    echo "Stopping $desc: $name "
    RETVAL=0
    if [ -n "$STRSS7_MKNOD" -a ":$STRSS7_REMOVEDEVICES" = ":yes" ] ; then
	echo -n "Removing STREAMS SS7 devices: "
	$STRSS7_MKNOD --remove
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    echo -n "Unloading STREAMS kernel modules: "
    modules=
    for module in $STRSS7_PRELOAD $STRSS7_DRIVERS $STRSS7_MODULES ; do
	modules="$module${modules:+ $modules}"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -r -q -- $module $redir"
	    if [ $? -ne 0 ] ; then
		echo -n "(failed) "
		RETVAL=1
	    fi
	fi
    done
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    return $RETVAL
}

restart() {
    stop
    start
    return $?
}

show() {
    echo "$name.sh: show: not yet implemented." >&2
    return 1
}

usage() {
    echo "Usage: /etc/init.d/$name.sh (start|stop|restart|force-reload|show)" >&2
    return 1
}

case "$1" in
    (start|stop|restart|show)
	$1 || RETVAL=$?
	;;
    (force-reload)
	restart || RETVAL=$?
	;;
    (*)
	usage || RETVAL=$?
	;;
esac

[ "${0##*/}" = "$name.sh" ] && exit $RETVAL

# =============================================================================
# 
# @(#) $RCSfile$ $Name$($Revision$) $Date$
#
# -----------------------------------------------------------------------------
#
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
#
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or modify it under
# the terms of the GNU Affero General Public License as published by the Free
# Software Foundation; version 3 of the License.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
# details.
#
# You should have received a copy of the GNU Affero General Public License along
# with this program.  If not, see <http://www.gnu.org/licenses/>, or write to
# the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
#
# -----------------------------------------------------------------------------
#
# U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
# behalf of the U.S. Government ("Government"), the following provisions apply
# to you.  If the Software is supplied by the Department of Defense ("DoD"), it
# is classified as "Commercial Computer Software" under paragraph 252.227-7014
# of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
# successor regulations) and the Government is acquiring only the license rights
# granted herein (the license rights customarily provided to non-Government
# users).  If the Software is supplied to any unit or agency of the Government
# other than DoD, it is classified as "Restricted Computer Software" and the
# Government's rights in the Software are defined in paragraph 52.227-19 of the
# Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
# the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
# (or any successor regulations).
#
# -----------------------------------------------------------------------------
#
# Commercial licensing and support of this software is available from OpenSS7
# Corporation at a fee.  See http://www.openss7.com/
#
# -----------------------------------------------------------------------------
#
# Last Modified $Date$ by $Author$
#
# -----------------------------------------------------------------------------
#
# $Log$
# Revision 0.9.2.8  2008-10-21 22:42:13  brian
# - handle verbose better in debian initscripts
#
# Revision 0.9.2.7  2008-04-29 07:11:34  brian
# - updating headers for release
#
# Revision 0.9.2.6  2007/08/12 16:49:59  brian
# - header updates
#
# =============================================================================


# vim: ft=sh sw=4 tw=80
